### DISCUSSION TITLE
What determine inference time?

### DISCUSSION THREAD

**Discussion by GMagician** [blakeblackshear/frigate#15539] [Ask A Question]
Hello, I see an inference high time variation, from 34ms up to 59ms with same Frigate config.
I'm using rknn detector + yolonas_m model + 2 cameras
Metrics report 0 detections on 1 camera and 0 or 1 on other.

What are the other things that determinate inference variation?

![{6A449094-A760-497D-8674-A06C182A950F}](https://github.com/user-attachments/assets/16ffa55c-e10f-4210-a669-6321da12b0d8)

**Comment by NickM-27** [comment]
It can depend on the activity of the resource doing the detection as well as system CPU usage in general 

Also depends on the specific model being used.

**Reply by GMagician** [reply]
ok but model is always the same and cpu resources is 8 up to 10%. Is this 2% enough to increase almost 100% inference time?

If I understood correctly how all this work, after a motion is detected (no model is involved to detect it) it's invoked the object detection (model now is involved) and inference is time spent by AI to identify what is moving on image. Am I right on this?

**Reply by NickM-27** [reply]
right, in this case the rockchip NPU is likely just having higher load causing the inference time to fluctuate

**Reply by GMagician** [reply]
The weird thing I don't understand is that it fluctuate over time but right now I'm writting with 0 detection on both cameras I'm expecting 0ms inference time, but it's 43ms. It's still working even when there are no detections? 
Sorry I bother you but I would like to understand how it works

**Reply by NickM-27** [reply]
the inference speed shown in the dashboard is just an average of the previous inferences

**Reply by GMagician** [reply]
And I continue to not understand. Now inference jumped to 57ms. No detection and even more no "motion box" on debug web page. I'm expecting a decreasing value in this situation, not increment.

**Reply by GMagician** [reply]
I see also a warning, on lower right window corner, with rknn is slow

**Reply by NickM-27** [reply]
Do you have any stationary objects currently detected?

**Reply by GMagician** [reply]
Where can I see it? in Debug page at "object list" tab is empty

**Reply by GMagician** [reply]
In debug I enabled all options and what I see different is the presente of the timestamp (but this is one of the options)

**Reply by NickM-27** [reply]
can you share screenshots of the system and camera metrics

**Reply by GMagician** [reply]
![{F6DDD7DB-A732-44B4-9DC6-8E9877E4F620}](https://github.com/user-attachments/assets/aecff4c9-4f67-4ace-8715-143e9e4afabb)

![{4BC6D4B2-DFDE-4683-8A1D-BF1721F8D5DE}](https://github.com/user-attachments/assets/f044e283-7772-44b7-ba3c-5d62592e3985)

**Reply by NickM-27** [reply]
what is your config?
